admin/libs/firebase_util.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const firebase = require('../../lib/firebase_handler.js').firebase;
+
+// Firebase utility class for managing search queries for an agent
+// (migrated from Datastore)
+const firebaseUtil = {
+  /**
+   * Gets a mock search entry point based on the agentId.
+   *
+   * @param {string} agentId The agent id with the associated search query.
```

Comment:
nit: This should be agentName since it's brand/123/agent/34234

Reply:
resolved